Satellite idm auth support 12891 (#1915)
theforeman/foreman-documentation: guides/common/modules/proc_using-IdM-credentials-to-log-in-to-the-Project-Web-UI-with-a-Chrome-browser.adoc

[id="Using_IdM_credentials_to_log_in_to_the_{Project}_Web_UI-with-a-Chrome-browser_{context}"]
= Using IdM credentials to log in to the {Project} Web UI with a Chrome browser

> This section describes how to use a Chrome browser to log in to your {Project} Web UI with your Identity Management (IdM) login + password.

Reviewer: same as above.

.Prerequisites

> * Your system administrator has enrolled your {Project} server into IdM and configured the server to use IdM for authentication. For more information, see {InstallingServerDocURL}Configuring_External_User_Groups_{project-context}#Using_FreeIPA_admin[Using Red Hat Identity Management] in _{InstallingServerDocTitle}_.

Reviewer: ```suggestion
* Your system administrator has enrolled your {Project} server into IdM and configured the server to use IdM for authentication.
For more information, see {InstallingServerDocURL}Configuring_External_User_Groups_{project-context}#Using_FreeIPA_admin[Using Red Hat Identity Management] in _{InstallingServerDocTitle}_.
```

link: same as above.

* An IdM server is running and reachable.

> * The host on which you are using the Chrome browser to log in to the {Project} Web UI is an IdM client.

Reviewer: Please change "{Project} Web UI" to "{ProjectWebUI}".

* You have a valid IdM login and password.
* [Recommended] You are using the latest version of the Chrome browser.

.Procedure
. Open a terminal window.
. Enable the Chrome browser to use Kerberos authentication:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
google-chrome –auth-server-whitelist="*._example.com_" --auth-negotiate-delegate-whitelist=”*._example.com_"
----

+
[NOTE]
====

> Instead of whitelisting the whole domain, you can also whitelist a specific {Project} Server.

Reviewer: ```suggestion
Instead of whitelisting the whole domain, you can also whitelist a specific {ProjectServer}.
```

====

. Obtain the Kerberos ticket-granting ticket (TGT) for yourself using your IdM credentials:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
kinit _idm_user_
Password for idm_user@_EXAMPLE.COM_:
----
. In your browser address bar, enter the URL of your {Project} Server.
+
You are logged in automatically.

[NOTE]
====

> Alternatively, you can skip the first three steps and enter your login + password in the fields displayed on the {Project} Web UI. This is also the only option if the host from which you are accessing the {Project} Web UI is not an IdM client.

Reviewer: Same as above. Please stick to "one sentence per line" as per our conventions.

> [id="Using_IdM_credentials_to_log_in_to_the_{Project}_Web_UI-with-a-Chrome-browser_{context}"]

Reviewer: ```suggestion
[id="Using_IdM_credentials_to_log_in_to_the_{ProjectWebUI}-with-a-Chrome-browser_{context}"]
```

Please either use `-` or `_` in your anchor. All file names must you dashes after the first underscore after the source file type (like `proc_`).

= Using IdM credentials to log in to the {ProjectWebUI} with a Chrome browser

This section describes how to use a Chrome browser to log in to your {ProjectWebUI} with your {FreeIPA} (IdM) login and password.

.Prerequisites
* Your system administrator has enrolled your {ProjectServer} into IdM and configured the server to use IdM for authentication.

> For more information, see {InstallingServerDocURL}Configuring_External_User_Groups_{project-context}#Using_FreeIPA_admin[Using {FreeIPA}] in _{InstallingServerDocTitle}_.

Reviewer: ```suggestion
For more information, see {InstallingServerDocURL}Configuring_External_User_Groups_{project-context}Using_FreeIPA_admin[Using {FreeIPA}] in _{InstallingServerDocTitle}_.
```

Author: Now that I look at the URL more closely, I believe it should be {InstallingServerDocURL}Configuring_External_User_Groups_{project-context}Using_FreeIPA_{project-context}[Using {FreeIPA}], should it not?

Reviewer: The current URL: https://docs.theforeman.org/nightly/Installing_Server/index-katello.html#Using_FreeIPA_foreman

which becomes `{InstallingServerDocURL}` + `Using_FreeIPA_` + `{project-context}` (instead of "foreman") + `[Using FreeIPA]` (title copied from the rendered document) + `in _{InstallingServerDocTitle}_`.

Author: Does `{InstallingServerDocURL}Using_FreeIPA_{project-context}[Using FreeIPA]` also work for the Satellite downstream URL, namely https://access.redhat.com/documentation/en-us/red_hat_satellite/6.12/html/installing_satellite_server_in_a_connected_network_environment/configuring_external_authentication_satellite#Using_FreeIPA_satellite?
AFAICT, this ^^ is a link to the same section in the multiple-html format, but it seems differently constructed; there is an extra middle section, namely "configuring_external_authentication_satellite". Am I missing something or is it indeed weird that the upstream and downstream versions are of different lengths

Author: I have tried the following URL, and it is working for both satellite and katello:
{InstallingServerDocURL}Using_FreeIPA_{project-context}[Using {FreeIPA}]. So you were right in the original comment. (But it's also nice to have curly brackets around FreeIPA.)

* An IdM server is running and reachable.
* The host on which you are using the Chrome browser to log in to the {ProjectWebUI} is an IdM client.
* You have a valid IdM login and password.
* {Team} recommends using the latest stable Chrome browser.

.Procedure

> . Open a terminal window.

Reviewer: ```suggestion
```

We generally don't add this to any (CLI) procedures.

. Enable the Chrome browser to use Kerberos authentication:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----

> google-chrome –auth-server-whitelist="*._example.com_" --auth-negotiate-delegate-whitelist=”*._example.com_"

Reviewer: ```suggestion
$ google-chrome –auth-server-whitelist="*._example.com_" --auth-negotiate-delegate-whitelist=”*._example.com_"
```

----

+
[NOTE]
====

> Instead of whitelisting the whole domain, you can also whitelist a specific {ProjectServer}.

Reviewer: Can we use "allowlisting" here? Maybe check what IBM style guide says about this and blocklisting.

Author: I cannot see 'whitelist' in the 2012 version but yes, it is there in the newest version.
I will change the word.
